src/librustdoc/html/static/main.js

```diff
@@ -90,6 +104,35 @@ if (!DOMTokenList.prototype.remove) {
         return null;
     }
 
+    function getCratesHash() {
+        if (window.hashCode === undefined || window.hashCode === 0) {
+            if (window.crates === undefined || window.crates.length === 0) {
+                window.crates = cloneArray(document.getElementById("crate-search")
+                                                   .getElementsByTagName("option"), 1)
+                                    .map(function(x) {
+                                        return x.value;
+                                    });
+                window.crates.sort();
+            }
+            window.hashCode = window.crates.join(",").hashCode();
+        }
+        return window.hashCode;
+    }
+
+    function getSavedCrates(savedCrates) {
+        if (typeof savedCrates === "undefined") {
```

Comment:
Why is this checking for undefined here? Why can't call sites just pass in `getCurrentValue(...)` as needed?

It looks like there's two call sites to this function: one passes getCurrentValue after a `!== null` check, the other calls this function with no arguments. The discrepancy there is a bit odd.

Reply:
This is a good point, this check is completely useless. Good catch!